chapters/equations.tex

```diff
@@ -103,15 +103,17 @@ \subsection{For-Equations -- Repetitive Equation Structures}\doublelabel{for-equ
 for IDENT in expression loop
 \end{lstlisting}
 
-The \lstinline!expression! of a for-equation shall be a vector expression. It is
-evaluated once for each for-equation, and is evaluated in the scope
+\subsubsection{Explicit Iteration Ranges of For-Equations}\doublelabel{explicit-iteration-ranges-of-for-equations}
+The \lstinline!expression! of a for-equation shall be a vector expression,
+where more general array expressions are treated as vector of vectors or vector of matrices.
+It is evaluated once for each for-equation, and is evaluated in the scope
 immediately enclosing the for-equation. The expression of a for-equation
 shall be a parameter expression. The iteration range of a for-equation
 can also be specified as Boolean or as an enumeration type, see
 \autoref{types-as-iteration-ranges} for more information. The loop-variable (\lstinline!IDENT!) is in scope
-inside the loop-construct and shall not be assigned to. The
-loop-variable has the same type as the type of the elements of the
-vector expression.
+inside the loop-construct and shall not be assigned to.
```

Comment:
This still seems to refer to the `expression` not being assigned to. But it should be that the `loop-variable` cannot be assigned to, right?

Reply:
I don't see that. The last sentence says ' The loop-variable is in scope inside the loop-construct and shall not be assigned to.' I don't see any other reference to not assigning to something.